pkg/controller/sparkapplication/submission.go

```diff
@@ -96,6 +96,9 @@ func buildSubmissionCommandArgs(app *v1beta2.SparkApplication, driverPodName str
 
 	args = append(args, "--master", masterURL)
 	args = append(args, "--deploy-mode", string(app.Spec.Mode))
+
+	// Add proxy user
+	args = append(args, addProxyUserOptions(app)...)
```

Comment:
Instead of having a separate method, you can simply do:

```
if app.Spec.ProxyUser != nil {
  args = append(args, ....)
}
```

Reply:
Yes, my original PR was like this. but i was asked to submit a test then i moved it to a separate method. I can revert it. Let me know.

Reply:
You can still unit test by checking the values of args.

Reply:
Updated the tests. Let me know if it's not okay.